utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -96,6 +94,26 @@ object BestEffortJsonSchemaEncoder {
     }
   }
 
+  private def encodeIntegerSchema(value: Any, schema: Schema, fieldName: Option[String] = None): Validated[NonEmptyList[String], Json] = {
+    def encodeBigDecimalToIntegerSchema(bigDecimalValue: BigDecimal): Validated[NonEmptyList[String], Json] = Try(bigDecimalValue.toLongExact) match {
+      case Failure(_) => error(value.getClass.getName, schema.toString, fieldName)
+      case Success(v) => Valid(jsonEncoder.encode(v))
+    }
+
+    value match {
+      case i: Int => Valid(jsonEncoder.encode(i))
+      case l: Long => Valid(jsonEncoder.encode(l))
+      case d: Double => encodeBigDecimalToIntegerSchema(BigDecimal.valueOf(d))
+      case f: Float => encodeBigDecimalToIntegerSchema(BigDecimal.valueOf(f.toDouble))
+      case bd: java.math.BigDecimal => encodeBigDecimalToIntegerSchema(bd)
+      case bd: scala.math.BigDecimal => encodeBigDecimalToIntegerSchema(bd)
+      case bi: scala.math.BigInt => encodeBigDecimalToIntegerSchema(BigDecimal(bi))
+      case bi: java.math.BigInteger => encodeBigDecimalToIntegerSchema(BigDecimal(bi))
+      case nm: Number => encodeBigDecimalToIntegerSchema(BigDecimal.valueOf(nm.doubleValue()))
```

Comment:
some people would do `new BigDecimal(nm.toString())` - though it may be not-so-fast

Reply:
`BigDecimal(nm.toString)`